summary/summary.py

```diff
@@ -0,0 +1,256 @@
+    writer.writerow(['removed'] + [len(summary[project]['removed']['count'] or []) for project in summary])
+    writer.writerow(['missing'] + [len(summary[project]['missing']['count'] or []) for project in summary])
+    writer.writerow([])
+    writer.writerow(['stubbed audited'] + [0 for project in summary])
+    writer.writerow(['removed audited'] + [0 for project in summary])
+    writer.writerow(['missing audited'] + [0 for project in summary])
+    writer.writerow([])
+    writer.writerow(['stubbed issues'] + [0 for project in summary])
+    writer.writerow(['removed issues'] + [0 for project in summary])
+    writer.writerow(['missing issues'] + [0 for project in summary])
+    writer.writerow([])
+    writer.writerow(['stubbed unaudited'] + [len(summary[project]['stubbed']['count'] or []) for project in summary])
+    writer.writerow(['removed unaudited'] + [len(summary[project]['removed']['count'] or []) for project in summary])
+    writer.writerow(['missing unaudited'] + [len(summary[project]['missing']['count'] or []) for project in summary])
+
+def chart_projects_json(summary):
+    result = {}
+
+    for project in summary:
+        data = summary[project]
+        result[project] = {
+            "stubbed": sorted(data['stubbed']['count']),
+            "removed": sorted(data['removed']['count']),
+            "missing": sorted(data['missing']['count']),
+            "stubbed-audited": [],
+            "removed-audited": [],
+            "missing-audited": [],
+            "stubbed-issues": [],
+            "removed-issues": [],
+            "missing-issues": [],
+            "stubbed-resolved": [],
+            "removed-resolved": [],
+            "missing-resolved": [],
+        }
+    return json.dumps(result, indent=2)
+
+################################################################
+#
+
+def main():
+    args = parse_arguments()
+    args.project = [os.path.normpath(project) for project in args.project]
+
+    configure_logger(args.verbose, args.debug)
+
+    with open(args.github, encoding="utf-8") as handle:
+        github = json.load(handle)
+
+    summary = {}
+
+    if not args.project:
+        args.project = list(github)
+
+    for project in args.project:
+        project_repo = github[project]['github']
+        proofs_path = github[project]['proofs']
+        project_path = os.path.basename(project_repo)
+        if args.clean:
+            clean_project(project_path)
+        if args.clone:
+            clone_project(project_repo)
+        if args.build:
+            build_project(project_path, proofs_path)
+        if args.summarize or args.chart:
+            result = summarize_project(project_path, proofs_path)
+            if result:
+                summary[project] = result
+
+    if args.summarize:
+        print(json.dumps(summary, indent=2))
+
+    if args.chart:
+#        chart_projects_csv(summary)
+        print(chart_projects_json(summary))
```

Comment:
why print?

Reply:
The semantics of the --summarize and --chart flags are to produce a summary or a chart.  How do you recommend producing them?